toolkits/confluence/arcade_confluence/client.py

```diff
@@ -0,0 +1,535 @@
+
+    async def post(self, path: str, **kwargs: Any) -> Any:
+        return await self.request("POST", path, **kwargs)
+
+    async def put(self, path: str, **kwargs: Any) -> Any:
+        return await self.request("PUT", path, **kwargs)
+
+
+class ConfluenceClientV1(ConfluenceClient):
+    def __init__(self, token: str):
+        super().__init__(token, api_version=ConfluenceAPIVersion.V1)
+
+    def construct_cql(
+        self, terms: list[str] | None, phrases: list[str] | None, enable_fuzzy: bool = False
+    ) -> str:
+        """Construct a CQL query from a list of terms and phrases.
+
+        Args:
+            terms: list of single words to search for
+            phrases: list of multiple words to search for
+            enable_fuzzy: enable fuzzy matching to find similar terms (e.g. 'roam' will find 'foam')
+
+        Learn about advanced searching using CQL here: https://developer.atlassian.com/cloud/confluence/advanced-searching-using-cql/
+        """
+        cql_parts = []
+
+        # Process single terms
+        if terms:
+            terms = split_by_space(terms)
+            term_queries = []
+            for term in terms:
+                term_suffix = "~" if enable_fuzzy else ""
+                term_queries.append(
+                    f'(text ~ "{term}{term_suffix}" OR title ~ "{term}{term_suffix}" OR space.title ~ "{term}{term_suffix}")'  # noqa: E501
+                )
+
+            if term_queries:
+                cql_parts.append(f"({' OR '.join(term_queries)})")
+
+        # Process phrases
+        if phrases:
+            phrase_queries = []
+            for phrase in phrases:
+                phrase_queries.append(
+                    f'(text ~ "{phrase}" OR title ~ "{phrase}" OR space.title ~ "{phrase}")'
+                )
+
+            if phrase_queries:
+                cql_parts.append(f"({' OR '.join(phrase_queries)})")
+
+        cql = " OR ".join(cql_parts)
+        return cql
+
+    def transform_search_content_response(
+        self, response: dict[str, Any]
+    ) -> dict[str, list[dict[str, Any]]]:
+        """
+        Transform the response from the GET /search endpoint by converting relative webui paths
+        to absolute URLs using the base URL from the response.
+        """
+        base_url = response.get("_links", {}).get("base", "")
+        transformed_results = []
+        for result in response.get("results", []):
+            content = result.get("content", {})
+            transformed_result = {
+                "id": content.get("id"),
+                "title": content.get("title"),
+                "type": content.get("type"),
+                "status": content.get("status"),
+                "excerpt": result.get("excerpt"),
+                "url": f"{base_url}{result.get('url')}",
+            }
+            transformed_results.append(transformed_result)
+
+        return {"results": transformed_results}
+
+
+class ConfluenceClientV2(ConfluenceClient):
```

Comment:
It looks to me the HTTP client is overloaded with stuff that belongs somewhere else, like transforming request and response payloads according to specific endpoints.

I don't see why we need two Client classes. Just instantiate the same client with a different `api_version`. The stuff that is not HTTP-related should ideally be outside the scope of the HTTP-client.